File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    update = []
    for y, f in enumerate(board):
        for x, value in enumerate(f):
            if value == 1:
                for a,b in [(x,y),(x-1,y-1),(x-1,y),(x-1,y+1),(x,y+1),(x,y-1),(x+1,y-1),(x+1,y),(x+1,y+1)]:
                    if 0 <= a <= len(board)-1 and 0 <= b <= len(board)-1 :
                        update.append((a,b))
    return len(board) * len(board) - len(list(set(update)))

File: 프로그래머스/0/120866. 안전지대/안전지대.java
class Solution {
    public int solution(int[][] board) {
    int n = board.length;
    int[] dx = {-1, -1, -1, 0, 0, 1, 1, 1};
    int[] dy = {-1, 0, 1, -1, 1, -1, 0, 1};
    int answer = 0;

    // 위험지역 표시
    for (int y = 0; y < n; y++) {
        for (int x = 0; x < n; x++) {
            if (board[y][x] == 1) {
                for (int i = 0; i < 8; i++) {
                    int ny = y + dy[i];
                    int nx = x + dx[i];
                    if (ny >= 0 && ny < n && nx >= 0 && nx < n && board[ny][nx] == 0) {
                        board[ny][nx] = 2; // 위험지역 표시
                    }
                }
            }
        }
    }

    // 안전한 지역 세기
    for (int y = 0; y < n; y++) {
        for (int x = 0; x < n; x++) {
            if (board[y][x] == 0) {
                answer++;
            }
        }
    }

    return answer;
}

}